TESTS/mbed_hal/stack_size_unification/main.cpp

```diff
@@ -31,19 +31,11 @@ extern osThreadAttr_t _main_thread_attr;
 #endif
 extern uint32_t mbed_stack_isr_size;
 
-#if !defined(MBED_CONF_RTOS_PRESENT)
-#define EXPECTED_ISR_STACK_SIZE                  (4096)
-#else
-#define EXPECTED_ISR_STACK_SIZE                  (1024)
-#endif
+#define EXPECTED_ISR_STACK_SIZE                  (MBED_CONF_TARGET_BOOT_STACK_SIZE)
 
-#if defined(TARGET_NUCLEO_F070RB) || defined(TARGET_STM32F072RB) || defined(TARGET_TMPM46B) || defined(TARGET_TMPM066)
-#define EXPECTED_MAIN_THREAD_STACK_SIZE          (3072)
-#else
-#define EXPECTED_MAIN_THREAD_STACK_SIZE          (4096)
-#endif
+#define EXPECTED_MAIN_THREAD_STACK_SIZE          (MBED_CONF_RTOS_MAIN_THREAD_STACK_SIZE)
 
-#define EXPECTED_USER_THREAD_DEFAULT_STACK_SIZE  (4096)
+#define EXPECTED_USER_THREAD_DEFAULT_STACK_SIZE  (MBED_CONF_RTOS_THREAD_STACK_SIZE)
```

Comment:
In the test case below, we use this macro to verify `OS_STACK_SIZE`:
```
TEST_ASSERT_EQUAL(EXPECTED_USER_THREAD_DEFAULT_STACK_SIZE, OS_STACK_SIZE);
```
where `OS_STACK_SIZE` is conditional
https://github.com/ARMmbed/mbed-os/blob/9a8c9e2c297f64c805698d72cd541ff3cd7fe538/rtos/source/TARGET_CORTEX/mbed_rtx_conf.h#L31-L35

The trouble is, to get the assertion to pass in *all* scenarios, our expectation needs to be conditional in the same way. But isn't it a duplication? I'm not sure there's any value in comparing two _macros_ in a _Greentea_ test to begin with...

Reply:
In my opinion, we should remove this assertion from the test case - which is out of this PR's initial scope (i.e. fixing line 34). But still this line of diff _doesn't make things worse_ than before (i.e. hardcoded 4096).

@0xc0170 If this PR blocks any release, maybe we can get it in as it is?